draft (#929)
percona/pmm-doc: docs/using/backup_mongo.md

## Restore a backup

### Restore compatibility

MongoDB backups can only be restored to the same service they were created from.

To restore a backup:

1. Go to <i class="uil uil-history"></i> **Backup > All backups** and find the backup that you want to restore.
2. Click the arrow in the **Actions** column to check all the information for the backup, then click ![](../_images/dots-three-vertical.png) **> Restore from backup**.
3. In the **Restore from backup** dialog, select **Same service** to restore to a service with identical properties or **Compatible services** to restore to a compatible service.
4. Select one of the available service names from the drop-down menu.
5. If you are restoring a PITR backup, also  select the point for the date and time that you want to restore the database to.
6. Check the values, then click **Restore**.
7. Go to the **Restores** tab to check the status of the restored backup.
During restoring, PMM disables all the scheduled backup tasks for the current service. Remember to re-enable them manually after the restore.

> ### Restore to a new cluster manually

Reviewer: Not sure about word **cluster**. We're supporting only replica sets. Cluster is more common term that includes also sharded setups.

Reviewer 2: per https://www.mongodb.com/basics/clusters
The Cluster's both ReplicaSet or Sharded.  
Since we have plans to add  Sharded, too - it's better to keep the generic name here

> 1. Install MongoDB and Percona Backup for MongoDB.

Reviewer: ```suggestion
1. Install MongoDB and Percona Backup for MongoDB. Pay attention to the versions. To minimize potential incompatibility use the same versions that were used for taking backups.
```

   For instructions, see the [PBM install documentation](https://docs.percona.com/percona-backup-mongodb/installation.html).
2. Configure your environment:

>      - to restore on a new environment with the same number of hosts and same replica names, make sure that the replica set names in your new destination cluster use the same names as those in the cluster that was backed up.<br/>

Reviewer: ```suggestion
     - to restore on a new environment with the same number of hosts and same replica set name, make sure that the replica set name in your new destination cluster use the same name as that in the cluster that was backed up.<br/>
```

      For more information, see [Restoring a backup into a new-environment](https://docs.percona.com/percona-backup-mongodb/usage/restore.html#restoring-a-backup-into-a-new-environment) in the PBM documentation. <br> 

>      - to restore **logical backups** to a new environment that has the same or more numbers of shards with different replica set names, configure the name mapping between the source and target environments. <br/>

Reviewer: ```suggestion
     - to restore **logical backups** to a new environment that has different replica set name, configure the name mapping between the source and target environments. <br/>
```

      For the new environment you can either set the **PBM_REPLSET_REMAPPING** environment variable for pbm CLI, or use the `--replset-remapping` flag for PBM commands.

        The mapping format is `<rsTarget>=<rsSource>`.

        For example:

>         `$ export PBM_REPLSET_REMAPPING="rsX=rsA,rsY=rsB"`

Reviewer: ```suggestion
        `$ export PBM_REPLSET_REMAPPING="targetRS=sourceRS"`
```

        OR 

>         `$ pbm restore <timestamp> --replset-remapping="rsX=rsA,rsY=rsB"`

Reviewer: ```suggestion
        `$ pbm restore <timestamp> --replset-remapping="targetRS=sourceRS"`
```

        For more information, see [Restoring into a replica set with a different name](https://docs.percona.com/percona-backup-mongodb/usage/restore.html#restoring-into-a-cluster-replica-set-with-a-different-name) in the PBM documentation.

3. Make sure that Percona Backup for MongoDB configuration in the new environment points to the remote storage defined for the original environment, including the authentication credentials for object stores.

    The easiest way to configure it is to create a config file, called, for example, `pbm_config.yaml`.

    For this, you can either copy the config from the source host or create a new one.

    To redirect config output from the existing environment, use `pbm config >> pbm_config.yaml`, then copy the resulting file to the new environment.

    Here's an example of config file content for AWS S3 compatible storage:

    ```yaml
    storage:
      type: s3
      s3:
        region: us-west-2
        bucket: pbm-test-bucket
        prefix: data/pbm/backup
        credentials:
          access-key-id: <your-access-key-id-here>
          secret-access-key: <your-secret-key-here> 
    ```

>     The prefix is the artifact name from the PMM **All Backups** page. 

Reviewer: ```suggestion
    The prefix is the **Backup name** from the PMM **All Backups** page. 
```

>     To implement the config, use the following command: `pbm config --file pbm_config.yaml`. 

Reviewer: We should wrap the command in the `code` tag.

          pbm list
            Backup snapshots: 2022-11-23T19:40:06Z [restore_to_time: 2021-01-13T15:53:40Z]
          ```
        b) provide the timestamp of the backup to the `pbm` command:

        `pbm restore 2022-11-23T19:40:06Z`

      For more information, see [Restore a backup](https://docs.percona.com/percona-backup-mongodb/usage/restore.html) topic in the PBM documentation.

    - For PITR backups:
        
        a) run the following command:

          ```code
            pbm list
            Backup snapshots:
              2022-11-23T19:40:06Z <logical> [restore_to_time: 2022-11-23T19:40:25Z]
              2022-11-23T19:45:07Z <logical> [restore_to_time: 2022-11-23T19:45:22Z]
            PITR <on>:
              2022-11-23T19:40:26Z - 2022-11-23T19:45:22Z
          ```
    b) provide the timestamp from one of the PITR ranges to the `pbm` command:

          `pbm restore --time="2022-11-23T19:40:26`
        
    For more information, see [Point-in-time Recovery topic in the PBM documentation](https://docs.percona.com/percona-backup-mongodb/usage/point-in-time-recovery.html).

6. Check the progress of the restore operation, using the following command:

>     `pbm list --restore`

Reviewer: ```suggestion
For logical restores:
    `pbm describe-restore <restore_name>`
    
For physical restores:
    `pbm describe-restore --config=/path/to/pbm_config.yaml <restore_name>`

Where <restore_name> will be given by PBM after starting the restoring, pbm_config.yaml - PBM config file you provided on the step 3.
```
